src/a-frame/three/data3d-view/io3d-material/fragment.glsl

```diff
@@ -40,8 +40,36 @@ void main() {
     #include <alphamap_fragment>
     #include <alphatest_fragment>
     #include <specularmap_fragment>
-    #include <normal_flip>
-    #include <normal_fragment>
+    //include <normal_fragment>
```

Comment:
wait... wasn't the idea to *use* `normal_fragment`?
It looks like you've inlined it here - was that on purpose?

Reply:
we were using both normal_flip and normal_fragment before as includes
now we are using normal_fragment CODE directly from the newest r87 for backwards compatability. Thats the only way we can support r84 and r87
The commented code was not clear, see my new commit.

Reply:
we can't just remove the `normal_flip` include, because then it will not work in r84 (because the r84 `normal_fragment` would be incomplete without it.
and we can't leave the `normal_flip` because r87 does not have it.

So my solution is to use r87 `normal_fragment` code in our shader

Reply:
Right, now the code matches the stated intent, so that's alright.

But one question remains: If I understand the r87 change log correctly, `normal_flip` was **merged into* `normal_fragment`. But I see that this is actually a smart move, because the tiny duplication fixes the backwards-compat issue, I guess?

Reply:
i'm not sure if i get your question
by three.js merging the two fragments it made all custom ShaderMaterials that use these includes faulty.
if you have include <normal_flip> in your code, it will not be able to compile the shader because <normal_flip> is not in three.js repository anymore but the old normal_fragment does not work without the normal_flip
i guess there is not many people using includes for their custom shaders.

Reply:
in short: using either of those includes is not compatible with all versions -> solution: don't use includes but use the actual shader code